Presentation template

Screedy opened the pull request:
- First version of the presentation template as issued in #1 
- Includes examples from my bc thesis which need to be deleted later

Screedy committed:
Create .gitignore

- Creation of .gitignore
- Added all unnecessary latex generated files
- Added mac metadata

Screedy committed:
Create UP_logo.png

Screedy committed:
Create prezentace.pdf

Screedy committed:
Create prezentace.tex

- Creation of basic template
- I left a few of my first slides from bc thesis presentation as an example

Tarasa24 commented:
Jestli máš zkušenosti s Github Actions, nechtěl bys rovnou vyrobit akci na generování PDF, vždycky když se tagne nová verze?Release by pak u sebe měl vygenerované PDF a nemuselo by tady takhle nevzhledně ležet

Tarasa24 reviewed (changes requested):
Jestli máš zkušenosti s Github Actions, nechtěl bys rovnou vyrobit akci na generování PDF, vždycky když se tagne nová verze?Release by pak u sebe měl vygenerované PDF a nemuselo by tady takhle nevzhledně ležet

Screedy committed:
Create latex-build.yml

- first version of automatic latex builder

Screedy committed:
Update latex-build.yml

- Changed full texlive to only base with needed extentions to reduce workflow run time.
- Changed paths

Screedy committed:
update output path

Screedy committed:
Language addition

- Added the czech language since its not in base latex.

Screedy committed:
testing the automation

Screedy committed:
Update latex-build.yml

Screedy committed:
Added the commit to repo

Commit:
Add generated Beamer PDF

Screedy commented:
@Tarasa24 automatické generování by už mělo fungovat

Tarasa24 commented on .github/workflows/latex-build.yml:
```diff
@@ -0,0 +1,49 @@
+          pdflatex -output-directory=presentation presentation/prezentace.tex
+          pdflatex -output-directory=presentation presentation/prezentace.tex
+
+      - name: Commit and push PDF file
+        run: |
+          git config --global user.name "github-actions"
+          git config --global user.email "github-actions@github.com"
+          git add presentation/prezentace.pdf
+          git commit -m "Add generated Beamer PDF"
+          git push
+
+      - name: Upload PDF artifact
+        uses: actions/upload-artifact@v3
+        with:
+          name: beamer-presentation
+          path: presentation/prezentace.pdf
```
Tady místo git push poprosím přes github akce vytvořit release a nahrát artefact do něj (pokud je to on tag).
Pokud je to on pull tak bych to dal možná jako comment do toho daného PR s odkazem to generované pdf

Tarasa24 commented on .github/workflows/latex-build.yml:
```diff
@@ -0,0 +1,49 @@
+name: Build Beamer Presentation
+
+on:
+  push:
+    branches:
+      - main
```
Místo přímo pushe bych dal on tag

Screedy replied:
```yml
on:
  push:
    branches:
      - main
    paths:
      - 'presentation/prezentace.tex'
```

Dal jsem to na změnu souboru, je to okey?

Screedy replied:
Už jsem to předělal na on tag.

Screedy committed:
Update latex-build.yml

Screedy committed:
Update prezentace.tex

Screedy committed:
Update latex-build.yml

Screedy committed:
Update prezentace.tex

Screedy committed:
Change to valid tag format

Screedy committed:
Update prezentace.tex

Screedy committed:
separate tag name generation

Screedy committed:
Update prezentace.tex

Screedy committed:
fixing warnings

- use of GITHUB_ENV and GITHUB_OUTPUT to ensure that tag name is accessible both as an environment var for subsequent steps and as an output for the action
- avoiding set-output since its deprecated

Screedy committed:
Update prezentace.tex

Screedy committed:
Changing version to avoid warning

Screedy committed:
Update prezentace.tex

Screedy committed:
Update latex-build.yml

Screedy committed:
Update prezentace.tex

Screedy committed:
use github api to avoid deprecation

Screedy committed:
Update prezentace.tex

Screedy committed:
Change to on trigger on tag

Screedy committed:
Debug options

Screedy committed:
Update latex-build.yml

Screedy committed:
Ignoring presentation pdf

- Added the prezentace.pdf to .gitignore for a cleaner look

Screedy committed:
Delete presentation/prezentace.pdf

Screedy committed:
Základní struktura

- změna uspořádání jmen autorů
- alespoň lehké přidání struktury prezentace

Screedy commented:
- [x] Změna z git push na on tag.
- [x] Změna pushe na nahrání artefaktu do release, který vyvolá vytvoření nového tagu.
- [x] Začal jsem aspoň okrajově moji část prezentace.
- [x] Přidal jsem úplně základní strukturu všech slajdů na kterých jsme se domluvili.
- [x] Přidání pdf prezentace do .gitignore.

Tarasa24 commented:
Najs díky, sorry že jsem se k tomu dostal až dneska
LGTM :tada:
